### PR TITLE
Add descriptions for queries H-Q

### PR DESCRIPTION
**Pull request by abdonpijpelink**
Relates to https://github.com/elastic/elasticsearch-specification/issues/964

Adds Query DSL descriptions for queries H through Q. Descriptions were sourced from https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html and subpages.

### CODE REVIEW

**Review by jrodewig — CHANGES_REQUESTED** [review]
Thanks for putting this up!

It looks like one of the parameter values is missing some text (https://github.com/elastic/elasticsearch-specification/pull/2213#discussion_r1279192146). I'm also not sure how we're handling formatting for fields without explicit keys (such as `<field>`).

Other than that, this looks good to me. I left several non-blocking nits that you can ignore if wanted.

File: specification/_types/query_dsl/fulltext.ts [hunk 1]
```diff
@@ -47,224 +47,654 @@ export class Intervals {
...
+   * You can use this suffix to use a different analysis method for exact matches.
+   */
   quote_field_suffix?: string
+  /**
+   * Method used to rewrite the query.
+   * @doc_id query-dsl-multi-term-rewrite
+   */
   rewrite?: MultiTermQueryRewrite
+  /**
+   * combining the several queries generated from the individual search terms is done using a `dis_max query` with the provided `tie_breaker`.
```
[the remark this answers is not in the document]
Comment (abdonpijpelink) [reply specification/_types/query_dsl/fulltext.ts:687]: I was struggling with this one and ended up forgetting to capitalize the "c". Still, it's not a good description. [The original docs](https://www.elastic.co/guide/en/elasticsearch/reference/master/query-dsl-query-string-query.html) don't have an explicit description for tie_breaker, but it is mentioned in the text:

_Since several queries are generated from the individual search terms, combining them is automatically done using a `dis_max` query with a `tie_breaker`._

I think I'll change it into:

_How to combine the queries generated from the individual search terms in the resulting `dis_max` query._

[the remark this answers is not in the document]
Comment (jrodewig) [reply specification/_types/query_dsl/fulltext.ts:687]: Sounds like a good fix. And yeah — Don't feel obligated to fix all our content debt here.

**Comment by abdonpijpelink** [comment]
Thanks for the thorough review @jrodewig . I've pushed a change that applies your suggestions.

**Review by jrodewig — APPROVED** [review]
Thanks!

**Pull request merged**